src/components/EditableTagField/EditableTagField.test.tsx

```diff
@@ -0,0 +1,163 @@
+import React from "react";
+import { v4 as uuid } from "uuid";
+import { render, fireEvent } from "test_utils/test-utils";
+import { EditableTagField } from ".";
+
+// Must mock uuid for this test since getRandomValues() is not supported in CI
+jest.mock("uuid");
+
+beforeAll(() => {
+  const MAX_INT = Number.MAX_SAFE_INTEGER;
+  uuid.mockImplementation(() =>
+    Math.floor(Math.random() * Math.floor(MAX_INT))
+  );
+});
+
+afterAll(() => jest.restoreAllMocks());
+
+const editableTags = [
+  { key: "keyA", value: "valueA" },
+  {
+    key: "keyB",
+    value: "valueB",
+  },
+  {
+    key: "keyC",
+    value: "valueC",
+  },
+];
+
+const defaultData = [...editableTags];
+
+test("Renders editable tags", async () => {
+  let data = [...defaultData];
+  const updateData = jest.fn((x) => {
+    data = x;
+  });
+
+  const { queryAllByDataCy, queryByText } = render(
+    <EditableTagField inputTags={editableTags} onChange={updateData} />
+  );
+
+  expect(queryAllByDataCy("user-tag-row")).toHaveLength(3);
+  expect(queryByText("hiddenField")).toBeNull();
+  expect(data).toEqual(defaultData);
+});
+
+test("Editing a tag value should update the tags", async () => {
+  let data = [...defaultData];
+  const updateData = jest.fn((x) => {
+    data = x;
+  });
+
+  const { queryAllByDataCy } = render(
+    <EditableTagField inputTags={editableTags} onChange={updateData} />
+  );
+
+  expect(data).toEqual(defaultData);
+  expect(queryAllByDataCy("user-tag-trash-icon")[0]).toBeVisible();
+
+  fireEvent.change(queryAllByDataCy("user-tag-value-field")[0], {
+    target: { value: "new value" },
```

Comment:
Can we add a test that edits the key? I know it is broken right now but I think it's important to have a test so that we know when it is working fully.

Reply:
https://github.com/evergreen-ci/spruce/pull/557/files/146a2e818d9a004b3fd1efbeb33daee0607db074#diff-2e9113b270269f5aa49f0dedf5c065d8a1b7064e76cd3364827f82cf92826310R95

There is a test that does that. As far as i can tell it works as expected.